File: mobile/lib/models/event.dart
import 'category.dart';

class Event {
  final int id;
  final int categoryId;
  final String title;
  final String? description;
  final String? organizer;
  final String? location;
  final String? eventDate;
  final String? startTime;
  final String? endTime;
  final int quota;
  final String? image;
  final String status;
  final int registeredCount;
  final Category? category;

  const Event({
    required this.id,
    required this.categoryId,
    required this.title,
    this.description,
    this.organizer,
    this.location,
    this.eventDate,
    this.startTime,
    this.endTime,
    required this.quota,
    this.image,
    required this.status,
    this.registeredCount = 0,
    this.category,
  });

  bool get isOpen => status == 'Aktif';
  bool get isFull => quota > 0 && registeredCount >= quota;

  factory Event.fromJson(Map<String, dynamic> json) {
    return Event(
      id: json['id'] as int,
      categoryId: (json['category_id'] as int?) ?? 0,
      title: json['title'] as String? ?? '',
      description: json['description'] as String?,
      organizer: json['organizer'] as String?,
      location: json['location'] as String?,
      eventDate: json['event_date'] as String?,
      startTime: json['start_time'] as String?,
      endTime: json['end_time'] as String?,
      quota: (json['quota'] as int?) ?? 0,
      image: json['image'] as String?,
      status: (json['status'] as String?) ?? 'Aktif',
      registeredCount: (json['registered_count'] as int?) ?? 0,
      category: json['category'] != null
          ? Category.fromJson(json['category'] as Map<String, dynamic>)
          : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'category_id': categoryId,
      'title': title,
      'description': description,
      'organizer': organizer,
      'location': location,
      'event_date': eventDate,
      'start_time': startTime,
      'end_time': endTime,
      'quota': quota,
      'image': image,
      'status': status,
    };
  }
}

File: backend/app/Models/Event.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Attributes\Fillable;
use Illuminate\Database\Eloquent\Attributes\Hidden;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;

#[Fillable([
    'category_id', 'title', 'description', 'organizer', 'location',
    'event_date', 'start_time', 'end_time', 'quota', 'image', 'status',
])]
#[Hidden(['created_at', 'updated_at'])]
class Event extends Model
{
    use HasFactory;

    public function category(): BelongsTo
    {
        return $this->belongsTo(Category::class);
    }

    public function registrations(): HasMany
    {
        return $this->hasMany(Registration::class);
    }
}
